src/rules/strictBooleanExpressionsRule.ts

```diff
@@ -0,0 +1,160 @@
+                if (!this.isBooleanType(lhsType) || !this.isBooleanType(rhsType)) {
+                    this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.BINARY_EXPRESSION_ERROR));
+                    errorFlagged = true;
+                }
+            } else {
+                this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.BINARY_EXPRESSION_ERROR));
+                errorFlagged = true;
+            }
+        }
+        if (!errorFlagged) {
+            super.visitBinaryExpression(node);
+        }
+    }
+
+    public visitPrefixUnaryExpression(node: ts.PrefixUnaryExpression) {
+        let errorFlagged = false;
+        let isExclamationOperator = node.operator === ts.SyntaxKind.ExclamationToken;
+        if (isExclamationOperator) {
+            let expr = node.operand;
+            let expType = this.checker.getTypeAtLocation(expr);
+            if (!this.isBooleanType(expType)) {
+                this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.UNARY_EXPRESSION_ERROR));
+                errorFlagged = true;
+            }
+        }
+        if (!errorFlagged) {
+            super.visitPrefixUnaryExpression(node);
+        }
+    }
+
+    public visitIfStatement(node: ts.IfStatement) {
+        let hasError = this.checkStatement(node);
+        if (!hasError) {
+            super.visitIfStatement(node);
+        }
+    }
+
+    public visitWhileStatement(node: ts.WhileStatement) {
+        let hasError = this.checkStatement(node);
+        if (!hasError) {
+            super.visitWhileStatement(node);
+        }
+    }
+
+    public visitDoStatement(node: ts.DoStatement) {
+        let hasError = this.checkStatement(node);
+        if (!hasError) {
+            super.visitDoStatement(node);
+        }
+    }
+
+    public visitConditionalExpression(node: ts.ConditionalExpression) {
+        let expType = this.checker.getTypeAtLocation(node.condition);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.CONDITIONAL_EXPRESSION_ERROR));
+        }
+        super.visitConditionalExpression(node);
+    }
+
+    public visitForStatement(node: ts.ForStatement) {
+        let expType = this.checker.getTypeAtLocation(node.condition);
+        if (!this.isBooleanType(expType)) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), `For ${Rule.STATEMENT_ERROR}`));
+        }
+        super.visitForStatement(node);
+    }
+
+    private checkStatement(node: StatementType): boolean {
+        let errorFlagged = false;
+        let expType = this.checker.getTypeAtLocation(node.expression);
+        if (!this.isBooleanType(expType)) {
+            switch (node.kind) {
+            case ts.SyntaxKind.IfStatement:
+                this.addFailure(this.createFailure(node.getStart(), node.getWidth(), `If ${Rule.STATEMENT_ERROR}`));
```

Comment:
Don't necessarily need different error message for each type of conditional, but this works